gumby/sync.py

```diff
@@ -235,8 +262,10 @@ def pushInfoToSubscribers(self):
         vars = {}
         for subscriber in self.connections_ready:
             subscriber_vars = subscriber.vars.copy()
-            subscriber_vars['port'] = subscriber.id + 12000
-            subscriber_vars['host'] = subscriber.transport.getPeer().host
+            if "port" not in subscriber_vars:
```

Comment:
If "port" is in subscriber vars, that means that the experiment client/node is explicitly telling the experiment server it is (going to be) using that port. With the old behaviour the experiment server is just ignoring what the client said and communicating it's thoughts on the matter to the other nodes. I'm unsure if it ever happened, but the old tribler experiment client used ports 21000 and up, so using a var[client_id]["port"] to construct a candidate should fail. At any rate, having the experiment server mess with my variables would not be something I'd expect as user of the vars-system.